internal/terraform/terraform_instance_config.go

```diff
@@ -0,0 +1,150 @@
+package terraform
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/go-logr/logr"
+	v0 "github.com/threeport/threeport/pkg/api/v0"
+	controller "github.com/threeport/threeport/pkg/controller/v0"
+	util "github.com/threeport/threeport/pkg/util/v0"
+)
+
+// TerraformInstanceConfig contains the configuration for terraform instance
+// reconciliation.
+type TerraformInstanceConfig struct {
+	r                   *controller.Reconciler
+	terraformInstance   *v0.TerraformInstance
+	terraformDefinition *v0.TerraformDefinition
+	log                 *logr.Logger
+	tfDirName           string
+	accessKeyId         string
+	secretAccessKey     string
+	tfState             string
+	tfOutput            string
+}
+
+// getTerraformInstanceOperations returns a list of operations for a terraform instance.
+func getTerraformInstanceOperations(c *TerraformInstanceConfig) *util.Operations {
+	operations := util.Operations{}
+
+	operations.AppendOperation(util.Operation{
+		Name:   "terraformInstance",
+		Create: c.createTerraformInstance,
+		Delete: c.deleteTerraformInstance,
+	})
+
+	return &operations
+}
+
+// createTerraformInstance runs the 'terraform apply' command to create
+// terraform-defined resources.
+func (c *TerraformInstanceConfig) createTerraformInstance() error {
+	// write terraform vars if applicable
+	tfVarsFilepath := fmt.Sprintf("%s/terraform.tfvars", c.tfDirName)
+	if c.terraformInstance.TerraformVarsDocument != nil {
+		if err := os.WriteFile(tfVarsFilepath, []byte(*c.terraformInstance.TerraformVarsDocument), 0644); err != nil {
+			return fmt.Errorf("failed to write terraform vars to file: %w", err)
+		}
+	}
+
+	// execute 'terrform apply'
+	applyCmd := exec.Command(
+		"terraform",
+		fmt.Sprintf("-chdir=%s", c.tfDirName),
+		"apply",
+		"-auto-approve",
+		"-no-color",
+	)
+	applyCmd.Env = append(
+		applyCmd.Environ(),
+		fmt.Sprintf("AWS_ACCESS_KEY_ID=%s", c.accessKeyId),
+		fmt.Sprintf("AWS_SECRET_ACCESS_KEY=%s", c.secretAccessKey),
+	)
```

Comment:
See if the terraform CLI supports an AWS_SESSION_TOKEN so that we can use short-lived creds.